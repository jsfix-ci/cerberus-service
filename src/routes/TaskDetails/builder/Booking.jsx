import React from 'react';
import {
  DATE_FORMATS,
  STRINGS,
} from '../../../utils/constants';
import {
  BookingUtil,
  PersonUtil,
  DateTimeUtil,
  MovementUtil,
} from '../../../utils';

import renderBlock from './helper/common';
import { calculateTimeDifference } from '../../../utils/Datetime/datetimeUtil';

const toBookingTimeDiference = (date, version) => {
  if (!DateTimeUtil.validate(date)) {
    return STRINGS.UNKNOWN_TEXT;
  }
  const journey = MovementUtil.movementJourney(version);
  const departureTime = MovementUtil.departureTime(journey);
  const dateTimeList = DateTimeUtil.toList(date, departureTime);
  return calculateTimeDifference(dateTimeList).replace(
    STRINGS.DAYJS_FUTURE,
    STRINGS.DAYJS_FUTURE_AIRPAX_REPLACE,
  );
};

const Booking = ({ version }) => {
  const booking = BookingUtil.get(version);
  const ticket = BookingUtil.bookingTicket(booking);
  const agent = BookingUtil.agent(booking);

  return (
    <div className="task-details-container">
      <h3 className="govuk-heading-m govuk-!-margin-top-0">Booking</h3>
      <div className="govuk-task-details-grid-column">
        {renderBlock('Reference', [BookingUtil.bookingRef(booking)])}
        {renderBlock('Number of travellers', [
          PersonUtil.totalPersons(version),
        ])}
        {renderBlock('Booking date', [
          DateTimeUtil.format(BookingUtil.bookedAt(booking), DATE_FORMATS.LONG),
          toBookingTimeDiference(BookingUtil.bookedAt(booking), version),
        ])}
        {renderBlock('Check-in date', [DateTimeUtil.format(BookingUtil.checkInAt(booking), DATE_FORMATS.LONG),
          toBookingTimeDiference(BookingUtil.checkInAt(booking), version)])}
        {renderBlock('Booking country', [
          `${BookingUtil.countryName(booking)} (${BookingUtil.countryCode(booking)})`,
        ])}
        {renderBlock('Booking type', [BookingUtil.bookingType(booking)])}
        {renderBlock('Ticket number', [BookingUtil.ticketNumber(ticket)])}
        {renderBlock('Ticket type', [BookingUtil.ticketType(ticket)])}
      </div>
      {BookingUtil.containsPayments(booking) ? (
        <>
          <div className="bottom-border-thin">
            <span className="font__light">Payments</span>
          </div>
          {BookingUtil.paymentsBlock(booking)}
        </>
      ) : <div className="bottom-border-thin" />}
      <div className="govuk-task-details-grid-column govuk-!-margin-top-3">
        {renderBlock('Agent IATA', [BookingUtil.agentIata(agent)])}
        {renderBlock('Agent location', [BookingUtil.agentLocation(agent)])}
      </div>
    </div>
  );
};

export default Booking;
