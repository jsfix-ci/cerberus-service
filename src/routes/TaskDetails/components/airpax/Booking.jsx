import React from 'react';
import {
  DATE_FORMATS,
  STRINGS,
} from '../../../../utils/constants';
import { BookingUtil,
  PersonUtil,
  DateTimeUtil,
  JourneyUtil } from '../../../../utils';

import renderBlock from '../../helper/common';
import { calculateTimeDifference } from '../../../../utils/Datetime/datetimeUtil';

// TODO: This is a duplicate of the one found in the roro Booking component.
const toBookingTimeDifference = (date, version) => {
  if (!DateTimeUtil.validate(date)) {
    return STRINGS.UNKNOWN_TEXT;
  }
  const journey = JourneyUtil.get(version);
  const departureTime = JourneyUtil.departureTime(journey);
  const dateTimeList = DateTimeUtil.toList(date, departureTime);
  return calculateTimeDifference(dateTimeList).replace(
    STRINGS.DAYJS_FUTURE,
    STRINGS.DAYJS_FUTURE_AIRPAX_REPLACE,
  );
};

const Booking = ({ version }) => {
  const booking = BookingUtil.get(version);
  const tickets = BookingUtil.bookingTickets(booking);
  const agent = BookingUtil.agent(booking);

  return (
    <div className="task-details-container">
      <h3 className="govuk-heading-m govuk-!-margin-top-0">Booking</h3>
      <div className="govuk-task-details-grid-column">
        {renderBlock('Reference', [BookingUtil.bookingRef(booking)])}
        {renderBlock('Number of travellers', [
          PersonUtil.totalPersons(version),
        ])}
        {renderBlock('Booking date', [
          DateTimeUtil.format(BookingUtil.bookedAt(booking), DATE_FORMATS.LONG),
          toBookingTimeDifference(BookingUtil.bookedAt(booking), version),
        ])}
        {renderBlock('Check-in date', [DateTimeUtil.format(BookingUtil.checkInAt(booking), DATE_FORMATS.LONG),
          toBookingTimeDifference(BookingUtil.checkInAt(booking), version)])}
        {renderBlock('Booking country', [
          `${BookingUtil.countryName(booking)} (${BookingUtil.countryCode(booking)})`,
        ])}
        {renderBlock('Booking type', [BookingUtil.bookingType(booking)])}
        {renderBlock('Ticket number', [BookingUtil.ticketNumbers(tickets)])}
        {renderBlock('Ticket type', [BookingUtil.ticketTypes(tickets)])}
      </div>
      {BookingUtil.containsPayments(booking) ? (
        <>
          <div className="bottom-border-thin">
            <span className="font__light">Payments</span>
          </div>
          {BookingUtil.paymentsBlock(booking)}
        </>
      ) : <div className="bottom-border-thin" />}
      <div className="govuk-task-details-grid-column govuk-!-margin-top-3">
        {renderBlock('Agent IATA', [BookingUtil.agentIata(agent)])}
        {renderBlock('Agent location', [BookingUtil.agentLocation(agent)])}
      </div>
    </div>
  );
};

export default Booking;
