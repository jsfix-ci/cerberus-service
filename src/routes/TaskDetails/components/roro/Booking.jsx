import React from 'react';
import classNames from 'classnames';

import { DATE_FORMATS, STRINGS } from '../../../../utils/constants';

// Utils
import { BookingUtil, DateTimeUtil } from '../../../../utils';
import renderBlock from '../../helper/common';
import { calculateTimeDifference, toDateTimeList, validateDate } from '../../../../utils/Datetime/datetimeUtil';
import { getDepartureTime, getJourney } from '../../../../utils/Journey/journeyUtil';

// TODO: This is a duplicate of the one found in the airpax Booking component.
const toBookingTimeDifference = (date, version) => {
  if (!validateDate(date)) {
    return STRINGS.UNKNOWN_TEXT;
  }
  const journey = getJourney(version);
  const departureTime = getDepartureTime(journey);
  const dateTimeList = toDateTimeList(date, departureTime);
  return calculateTimeDifference(dateTimeList).replace(
    STRINGS.DAYJS_FUTURE,
    STRINGS.DAYJS_FUTURE_AIRPAX_REPLACE,
  );
};

const Booking = ({ version, classModifiers }) => {
  const booking = BookingUtil.get(version);
  const tickets = BookingUtil.bookingTickets(booking);
  return (
    <div className={classNames('task-details-container', classModifiers)}>
      <h3 className="govuk-heading-m govuk-!-margin-top-0">Booking and check-in</h3>
      <div className="govuk-task-details-grid-column">
        {renderBlock('Reference', [BookingUtil.bookingRef(booking)])}
        {renderBlock('Ticket number', [BookingUtil.ticketNumbers(tickets)])}
        {renderBlock('Type', [BookingUtil.ticketTypes(tickets)])}
        {renderBlock('Name', [BookingUtil.bookingName(booking)])}
        {renderBlock('Address', [BookingUtil.bookingAddress(booking)])}
        {renderBlock('Booking date and time', [
          DateTimeUtil.format(BookingUtil.bookedAt(booking), DATE_FORMATS.LONG),
          toBookingTimeDifference(BookingUtil.bookedAt(booking), version),
        ])}
        {renderBlock('Country', [
          `${BookingUtil.countryName(booking)} (${BookingUtil.countryCode(booking)})`,
        ])}
        {renderBlock('Payment method', [BookingUtil.paymentMethod(booking)])}
        {renderBlock('Ticket price', [BookingUtil.ticketPrices(tickets)])}
        {renderBlock('Ticket type', [BookingUtil.ticketTypes(tickets)])}
        {renderBlock('Check-in date and time', [DateTimeUtil.format(BookingUtil.checkInAt(booking), DATE_FORMATS.LONG),
          toBookingTimeDifference(BookingUtil.checkInAt(booking), version)])}
      </div>
    </div>
  );
};

export default Booking;
