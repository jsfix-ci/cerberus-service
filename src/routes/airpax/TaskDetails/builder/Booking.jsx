import React from 'react';
import {
  LONG_DATE_FORMAT,
  DAYJS_FUTURE,
  DAYJS_FUTURE_AIXPAX_REPLACE,
} from '../../../../constants';
import {
  BookingUtil,
  PersonUtil,
  DateTimeUtil,
  MovementUtil,
} from '../../utils';

import renderBlock from './helper/common';
import calculateTimeDifference from '../../../../utils/calculateDatetimeDifference';

const toBookingTimeDiference = (booking, version) => {
  const journey = MovementUtil.movementJourney(version);
  const departureTime = MovementUtil.departureTime(journey);
  const dateTimeList = DateTimeUtil.toList(
    BookingUtil.bookedAt(booking),
    departureTime,
  );
  return calculateTimeDifference(dateTimeList).replace(
    DAYJS_FUTURE,
    DAYJS_FUTURE_AIXPAX_REPLACE,
  );
};

const Booking = ({ version }) => {
  const booking = BookingUtil.get(version);
  const ticket = BookingUtil.bookingTicket(booking);
  const agent = BookingUtil.agent(booking);

  return (
    <div className="task-details-container">
      <h3 className="govuk-heading-m govuk-!-margin-top-0">Booking</h3>
      <div className="govuk-task-details-grid-column">
        {renderBlock('Reference', [BookingUtil.bookingRef(booking)])}
        {renderBlock('Number of travellers', [
          PersonUtil.totalPersons(version),
        ])}
        {renderBlock('Booking date', [
          DateTimeUtil.format(BookingUtil.bookedAt(booking), LONG_DATE_FORMAT),
          toBookingTimeDiference(booking, version),
        ])}
        {renderBlock(undefined, undefined)}
        {renderBlock('Booking country', [
          `${BookingUtil.countryName(booking)} (${BookingUtil.countryCode(
            booking,
          )})`,
        ])}
        {renderBlock('Booking type', [BookingUtil.bookingType(booking)])}
        {renderBlock('Ticket number', [BookingUtil.ticketNumber(ticket)])}
        {renderBlock('Ticket type', [BookingUtil.ticketType(ticket)])}
      </div>
      {BookingUtil.containsPayments(booking) ? (
        <>
          <div className="bottom-border-thin">
            <span className="font__light">Payments</span>
          </div>
          {BookingUtil.paymentsBlock(booking)}
        </>
      ) : <div className="bottom-border-thin" />}
      <div className="govuk-task-details-grid-column govuk-!-margin-top-3">
        {renderBlock('Agent IATA', [BookingUtil.agentIata(agent)])}
        {renderBlock('Agent location', [BookingUtil.agentLocation(agent)])}
      </div>
    </div>
  );
};

export default Booking;
