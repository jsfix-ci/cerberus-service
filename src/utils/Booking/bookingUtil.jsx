import React from 'react';
import lookup from 'country-code-lookup';

import { DATE_FORMATS, STRINGS, UNITS } from '../constants';

import CommonUtil from '../Common/commonUtil';
import { calculateTimeDifference, getFormattedDate, toDateTimeList } from '../Datetime/datetimeUtil';
import { formatField } from '../FieldFormat/fieldFormatterUtil';

const getPaymentMethod = (booking) => {
  if (!booking) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return booking?.paymentMethod || STRINGS.UNKNOWN_TEXT;
};

// TODO: Is there an address against the booking?
const getBookingAddress = (booking) => {
  if (!booking) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return STRINGS.UNKNOWN_TEXT;
};

// TODO: Is there a name attached to the booking or is the main person attached to the booking?
const getBookingName = (booking) => {
  if (!booking) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return STRINGS.UNKNOWN_TEXT;
};

const getAgentLocation = (agent) => {
  if (!agent?.location) {
    return STRINGS.UNKNOWN_TEXT;
  }
  const splitLoc = agent?.location?.split(',').map((val) => val.trim());
  if (!splitLoc.every((val) => val)) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return [splitLoc[0], CommonUtil.iso3Code(splitLoc[1])].join(', ');
};

const getAgentIata = (agent) => {
  if (!agent?.iata) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return agent.iata;
};

const getAgentName = (agent) => {
  if (!agent?.name) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return agent.name;
};

const hasAgent = (booking) => {
  return !!booking?.agent;
};

const getAgent = (booking) => {
  if (hasAgent(booking)) {
    return booking.agent;
  }
  return null;
};

const hasPaymentCard = (payment) => {
  return !!payment?.card;
};

const getPaymentCard = (payment) => {
  if (hasPaymentCard(payment)) {
    return payment.card;
  }
  return null;
};

const getCardExpiry = (payment) => {
  const paymentCard = getPaymentCard(payment);
  if (paymentCard?.expiry) {
    return getFormattedDate(paymentCard.expiry, DATE_FORMATS.CUSTOM_CARD_EXPIRY);
  }
  return STRINGS.UNKNOWN_TEXT;
};

const getCardLastFourDigits = (payment) => {
  const paymentCard = getPaymentCard(payment);
  if (paymentCard?.number) {
    return paymentCard.number.substr(paymentCard.number.length - 4);
  }
  return STRINGS.UNKNOWN_TEXT;
};

const getPaymentAmount = (payment) => {
  if (!payment?.amount) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return payment.amount;
};

const hasPayments = (booking) => {
  return !!booking?.payments?.length > 0;
};

const getPayments = (booking) => {
  if (hasPayments(booking)) {
    return booking.payments;
  }
  return null;
};

const getTicketPrices = (tickets) => {
  if (!tickets?.length) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return tickets
    .filter((ticket) => !!ticket.price)
    .map((ticket) => {
      if (ticket.price.startsWith(UNITS.CURRENCY.unit)) {
        return ticket.price;
      }
      return formatField(UNITS.CURRENCY.name, ticket.price);
    }).join(', ') || STRINGS.UNKNOWN_TEXT;
};

const getTicketTypes = (tickets) => {
  if (!tickets?.length) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return tickets.filter((ticket) => !!ticket.type).map((ticket) => ticket.type).join(', ') || STRINGS.UNKNOWN_TEXT;
};

const getTicketNumbers = (tickets) => {
  if (!tickets?.length) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return tickets.filter((ticket) => !!ticket?.number).map((ticket) => ticket?.number).join(', ') || STRINGS.UNKNOWN_TEXT;
};

const hasTickets = (booking) => {
  return !!booking?.tickets?.length;
};

const getTickets = (booking) => {
  if (hasTickets(booking)) {
    return booking.tickets;
  }
  return null;
};

const getBookingType = (booking) => {
  if (!booking?.type) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return booking.type;
};

const getBookingCountryCode = (booking) => {
  if (!booking?.country) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return CommonUtil.iso3Code(booking.country);
};

const getBookingCountryName = (booking) => {
  const countryCode = getBookingCountryCode(booking);
  if (countryCode === STRINGS.UNKNOWN_TEXT) {
    return STRINGS.UNKNOWN_TEXT;
  }
  if (lookup.byIso(countryCode) === null) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return lookup.byIso(countryCode).country;
};

const toBookingDateText = (booking) => {
  if (!booking?.bookedAt) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return getFormattedDate(booking.bookedAt, DATE_FORMATS.SHORT_ALT);
};

const getBookingReference = (booking) => {
  if (!booking?.reference) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return booking.reference;
};

const getCheckInAt = (booking) => {
  if (!booking?.checkInAt) {
    return STRINGS.UNKNOWN_TEXT;
  }
  return booking.checkInAt;
};

const toCheckInTimeText = (booking) => {
  const checkinPrefix = 'Check-in';
  if (!booking?.checkInAt) {
    return `${checkinPrefix} ${STRINGS.UNKNOWN_TEXT}`;
  }
  return `${checkinPrefix} ${getFormattedDate(booking.checkInAt, DATE_FORMATS.CUSTOM_HOUR_MINUTE)}`;
};

const getBookedAt = (booking) => {
  return booking?.bookedAt;
};

const getBookedPriorToDeparture = (bookedAt, departureTime) => {
  const dateTimeList = toDateTimeList(bookedAt, departureTime);
  return calculateTimeDifference(dateTimeList);
};

const hasBooking = (targetTask) => {
  return !!targetTask?.movement?.booking;
};

const getBooking = (targetTask) => {
  if (hasBooking(targetTask)) {
    return targetTask.movement.booking;
  }
  return null;
};

const toPaymentsBlock = (booking) => {
  const payments = getPayments(booking);
  if (payments) {
    return payments.map((payment, index) => {
      return (
        <div key={index} className="bottom-border-thin govuk-!-margin-top-1 govuk-!-padding-bottom-1">
          <div className="font__bold">{getPaymentAmount(payment)}</div>
          <div className="font__light">
            Credit card ending {getCardLastFourDigits(payment)}, expiry {getCardExpiry(payment)}
          </div>
        </div>
      );
    });
  }
};

const BookingUtil = {
  get: getBooking,
  bookedAt: getBookedAt,
  checkInAt: getCheckInAt,
  bookedPrior: getBookedPriorToDeparture,
  bookingAddress: getBookingAddress, // TODO: See comment above
  bookingName: getBookingName, // TODO: See comment above
  toCheckInText: toCheckInTimeText,
  bookingRef: getBookingReference,
  toBookingText: toBookingDateText,
  countryCode: getBookingCountryCode,
  countryName: getBookingCountryName,
  bookingType: getBookingType,
  bookingTickets: getTickets,
  ticketNumbers: getTicketNumbers,
  ticketTypes: getTicketTypes,
  ticketPrices: getTicketPrices,
  payments: getPayments,
  containsPayments: hasPayments,
  paymentAmount: getPaymentAmount,
  paymentCard: getPaymentCard,
  cardLastFourDigits: getCardLastFourDigits,
  cardExpiry: getCardExpiry,
  agent: getAgent,
  agentName: getAgentName,
  agentIata: getAgentIata,
  agentLocation: getAgentLocation,
  paymentsBlock: toPaymentsBlock,
  paymentMethod: getPaymentMethod,
};

export default BookingUtil;

export {
  toBookingDateText,
  getBookingReference,
  toCheckInTimeText,
  getBooking,
  getBookedAt,
  getCheckInAt,
  getBookedPriorToDeparture,
  getBookingCountryCode,
  getBookingCountryName,
  getBookingType,
  getTickets,
  getTicketNumbers,
  getTicketTypes,
  getTicketPrices,
  getPayments,
  getPaymentAmount,
  getPaymentCard,
  getCardLastFourDigits,
  getCardExpiry,
  getAgent,
  getAgentName,
  getAgentIata,
  getAgentLocation,
  toPaymentsBlock,
};
