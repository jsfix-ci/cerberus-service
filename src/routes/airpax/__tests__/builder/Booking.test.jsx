import React from 'react';
import renderer from 'react-test-renderer';

import Booking from '../../TaskDetails/builder/Booking';
import taskDetailsData from '../../__fixtures__/taskData_AirPax_TaskDetails.fixture.json';

describe('Booking', () => {
  it('should render the booking component', () => {
    const tree = renderer.create(<Booking version={taskDetailsData.versions[1]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
