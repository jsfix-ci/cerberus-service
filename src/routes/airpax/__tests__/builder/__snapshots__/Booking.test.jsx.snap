// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking should render the booking component 1`] = `
<div
  className="task-details-container"
>
  <h3
    className="govuk-heading-m govuk-!-margin-top-0"
  >
    Booking
  </h3>
  <div
    className="govuk-task-details-grid-column"
  >
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Reference
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        LSV4UV
      </p>
    </div>
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Number of travellers
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        4
      </p>
    </div>
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Booking date
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        Unknown
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Unknown
      </p>
    </div>
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Booking country
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        Unknown (Unknown)
      </p>
    </div>
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Booking type
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        Unknown
      </p>
    </div>
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Ticket number
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        Unknown
      </p>
    </div>
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Ticket type
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        Unknown
      </p>
    </div>
  </div>
  <div
    className="bottom-border-thin"
  >
    <span
      className="font__light"
    >
      Payments
    </span>
  </div>
  <div
    className="bottom-border-thin govuk-!-margin-top-1 govuk-!-padding-bottom-1"
  >
    <div
      className="font__bold"
    >
      2190.48
    </div>
    <div
      className="font__light"
    >
      Credit card ending 
      X63X
      , expiry 
      10/20
    </div>
  </div>
  <div
    className="bottom-border-thin govuk-!-margin-top-1 govuk-!-padding-bottom-1"
  >
    <div
      className="font__bold"
    >
      2190.48
    </div>
    <div
      className="font__light"
    >
      Credit card ending 
      X63X
      , expiry 
      10/20
    </div>
  </div>
  <div
    className="govuk-task-details-grid-column govuk-!-margin-top-3"
  >
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Agent IATA
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        Unknown
      </p>
    </div>
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Agent location
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        Unknown
      </p>
    </div>
  </div>
</div>
`;
